### DISCUSSION TITLE
Editor for Button Layout Screen on Display in the Webconfig

### DISCUSSION THREAD

**Discussion by pvtex** [OpenStickCommunity/GP2040-CE#1211] [Ideas]
Hello,

would be great to edit the graphical Layout of the buttons on the display in the webconfig.
Could be very rudimental.
 - how many buttons
 - for each button
   - type
   - size
   - position
   - button assign

would make it much easier to implement new control surfaces without recompiling each time to check the correct button placement.

rergards,
Bjoern

**Comment by mikepparks** [comment]
This is something we already have in various states of progress, but nothing solid yet. It will happen, just a matter of when.

**Comment by nxarcadepl** [comment]
is there any way to set my own button layout for today?? I have an unusual Mortal Kombat + Street Fighter layout and I would like to set my own layout :)

![HolzPAD](https://github.com/user-attachments/assets/0abe4c91-9c3f-4680-99cd-0296eadc6f08)

**Reply by TheTrainGoes** [reply]
@nxarcadepl you can compile your own firmware with a custom layout as defined in the boardconfig.  You can use the Zero Rhythm as a template example.
